v3/src/data-interactive/data-interactive-utils.ts

```diff
@@ -13,3 +16,19 @@ export function canonicalizeAttributeName(name: string, iCanonicalize = true) {
   }
   return tName
 }
+
+export function getCaseValues(caseId: string, collectionId: string, dataSet: IDataSet) {
+  const attributes = dataSet.getGroupedCollection(collectionId)?.attributes ??
+    dataSet.ungroupedAttributes
+
+  const values: DICaseValues = {}
+  const actualCaseIndex = dataSet.cases.findIndex(actualCase => actualCase.__id__ === caseId)
```

Comment:
```suggestion
  const actualCaseIndex = dataSet.caseIDMap.get(caseId)
```
Avoids potential O(n^2) lookup.